bundles/org.openhab.binding.carnet/src/main/java/org/openhab/binding/carnet/internal/handler/CarNetVehicleHandler.java

```diff
@@ -0,0 +1,643 @@
+                    action = switchOn ? "startWindowHeat" : "stopWindowHeat";
+                    actionStatus = api.controlWindowHeating(switchOn);
+                    break;
+                case CHANNEL_CONTROL_PREHEAT:
+                    sendOffOnError = true;
+                    action = switchOn ? "startPreHeat" : "stopPreHeat";
+                    actionStatus = api.controlPreHeating(switchOn);
+                    break;
+                case CHANNEL_CONTROL_VENT:
+                    sendOffOnError = true;
+                    action = switchOn ? "startVentilation" : "stopVentilation";
+                    actionStatus = api.controlVentilation(switchOn, 15);
+                    break;
+                default:
+                    logger.info("{}: Channel {} is unknown, command {} ignored", thingId, channelId, command);
+                    break;
+            }
+
+            updateActionStatus(action, actionStatus);
+            forceUpdate = true; // update on successful command
+        } catch (CarNetException e) {
+            CarNetApiErrorDTO res = e.getApiResult().getApiError();
+            if (res.isOpAlreadyInProgress()) {
+                logger.warn("{}: \"An operation is already in progress, request was rejected!\"", thingId);
+            } else {
+                error = getError(e);
+                logger.warn("{}: {}", thingId, error.toString());
+            }
+        } catch (RuntimeException e) {
+            error = "General Error: " + getString(e.getMessage());
+            logger.warn("{}: {}", thingId, error, e);
+        }
+
+        if (!error.isEmpty()) {
+            if (sendOffOnError) {
+                updateState(channelUID.getId(), OnOffType.OFF);
+            }
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, error);
+        }
+    }
+
+    private void updateActionStatus(String action, String actionStatus) {
+        if (!actionStatus.isEmpty()) {
+            if (!action.isEmpty()) {
+                updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_ACTION, getStringType(action));
+            }
+            updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_ACTION_STATUS, getStringType(actionStatus));
+            boolean inProgress = actionStatus.equalsIgnoreCase(CNAPI_REQUEST_IN_PROGRESS);
+            updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_ACTION_PENDING,
+                    inProgress ? OnOffType.ON : OnOffType.OFF);
+            updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_UPDATED, getTimestamp());
+        }
+    }
+
+    /**
+     * This routine is called every time the Thing configuration has been changed (e.g. PaperUI)
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("{}: Thing config updated.", thingId);
+        super.handleConfigurationUpdate(configurationParameters);
+        initializeThing();
+    }
```

Comment:
Is this really necessary? The framework calls `initialize()` when the configuration changed.